libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/api/MapboxTurnIconsApi.kt

```diff
@@ -0,0 +1,59 @@
+package com.mapbox.navigation.ui.maneuver.api
+
+import com.mapbox.bindgen.Expected
+import com.mapbox.bindgen.ExpectedFactory
+import com.mapbox.navigation.base.internal.maneuver.ManeuverTurnIcon
+import com.mapbox.navigation.ui.maneuver.TurnIconHelper
+import com.mapbox.navigation.ui.maneuver.model.TurnIconError
+import com.mapbox.navigation.ui.maneuver.model.TurnIconResources
+import com.mapbox.navigation.ui.utils.internal.ifNonNull
+
+/**
+ * An API for generating turn icon drawable resource references.
+ *
+ * @param turnIconResources a [TurnIconResources] reference
+ */
+class MapboxTurnIconsApi(var turnIconResources: TurnIconResources) {
+
+    private var turnIconHelper = TurnIconHelper(turnIconResources)
+
+    /**
+     * Returns a reference to a drawable for turn icons.
+     *
+     * @param type a maneuver type
+     * @param degrees a value describing the degree of turn
+     * @param modifier a [ManeuverModifier] value
+     * @param drivingSide the driving side describing the turn
+     */
+    fun generateTurnIcon(
```

Comment:
Will you also migrate the `MapboxTurnIconManeuver` class to use this API?

That way our primary use case can ensure this api works

Reply:
sure